src/groupByKey.ts

```diff
@@ -2,6 +2,29 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<ObjectToSort, Key extends keyof ObjectToSort>(
+  items: ObjectToSort[],
+  key: Key,
+): GroupsMap<ObjectToSort> {
+  if (items.length === 0) {
+    return {};
+  }
+
+  type ValueFromObject = ObjectToSort[Key];
+
+  const valuesOfKey: Array<ValueFromObject> = items.map(
```

Comment:
The map operation in line 16 is unnecessary. You're creating an array of values only to create a set later. This is inefficient as it results in two iterations over the items array.